src/wasm-linker.cpp

```diff
@@ -297,114 +298,38 @@ void Linker::emscriptenGlue(std::ostream& o) {
     WasmPrinter::printModule(&out.wasm, std::cerr);
   }
 
-  out.wasm.removeImport(EMSCRIPTEN_ASM_CONST); // we create _sig versions
-
-  makeDynCallThunks();
-
-  o << ";; METADATA: { ";
-  // find asmConst calls, and emit their metadata
-  struct AsmConstWalker : public PostWalker<AsmConstWalker, Visitor<AsmConstWalker>> {
-    Linker* parent;
-
-    std::map<std::string, std::set<std::string>> sigsForCode;
-    std::map<std::string, Address> ids;
-    std::set<std::string> allSigs;
+  for (auto f : emscripten::makeDynCallThunks(out.wasm, getTableData())) {
```

Comment:
Rephrasing: we don't need it, but I think it's cleaner.

Previously we had a check that `size > 0` so we would avoid calling `getTableSegment`, which would add a segment if there was none, which would cause an empty table to be emitted in the wasm, vs no table which was expected.
The progression was essentially: check or misbehave quietly -> check or fail loudly -> don't need to check. So, the whole point of the `getTableData` method is to abstract out the `if`. Whether that's a *worthwhile* abstraction, I'm not totally sure, though I feel the calling code is clearer.